pkg/tools/custom_tool.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tools
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/GoogleCloudPlatform/kubectl-ai/gollm"
+)
+
+// CustomToolConfig defines the structure for configuring a custom tool.
+type CustomToolConfig struct {
+	Name    string                    `yaml:"name"`
+	FnDef   *gollm.FunctionDefinition `yaml:"fndef"`
+	Command []string                  `yaml:"command"` // Command and args to execute
+}
```

Comment:
would be great if you can share an example of what a real config look like say for `nomos` or `helm`

Better will be a quick screencast showing this in action to get a feel for the UX.

Reply:
Sure, will add an example or demo once I've tested this out.

Reply:
@droot Updated the example config and screenshot in https://github.com/GoogleCloudPlatform/kubectl-ai/pull/223#issue-3058426043

Reply:
@droot Do we need both the tool description and command description?

Reply:
Awesome. Taking a look soon.